code/API_definitions/device-status.yaml

```diff
@@ -588,12 +777,50 @@ components:
     ROAMING_STATUS:
       value:
         subscriptionId: 123654
-        eventType: ROAMING_STATUS
-        eventTime: 2023-01-17T13:18:23.682Z
-        eventDetail:
-          ueId:
-            msisdn: 123456789
-          roaming: true
-          countryCode: 208
-          countryName: 
-            FR 
+        event:
+          eventType: ROAMING_STATUS
+          eventTime: 2023-01-17T13:18:23.682Z
+          eventDetail:
+            device:
+              phoneNumber: 123456789
```

Comment:
The example given for the events all include phoneNumber as example of the identification of the device. When we set up a subscription there are several ways to identify the device for which we want to subscribe to device status. All those being optional. 

I would assume that whatever you input as an identifier is also what you will receive back with the event. The reason for that would be that if the application have not been trusted with knowing the actual phoneNumber of the device this should not be exposed when calling the Device Status API. Is this a common understanding and in that case should it be mentioned in the documentation of the API?

Reply:
Yes, that is my understanding - the API should return the at most same device identifiers as were provided by the client, but could maybe remove some if they were not valid or not used by the API provider.

For example, if the API client provides both `phoneNumber`  and `ipv4Address`, but of those two the implementation only uses `phoneNumber` to identify the device, then only `phoneNumber` should be included in responses. But that is my view - it would not be wrong to return `ipv4Address` as well even if that could be interpreted as confirming that the device is using that IP address when that has not been checked.

My view is that returning these values also gives the API provider an opportunity to "clean up" the formatting to show the API client the preferred field formatting. As we tighten up format patterns, there is less scope for the API client to use formats that are difficult to parse but, for example, if the client provided `+123456789` and the API returned `123456789`, that would indicate that we would prefer the optional `+` not to be used.

Reply:
Maybe a topic for the TSC, but the point is that the current definition does allow the API consumer to provide more than one device identifier. This is inherited from QoD, where the motivation to allow more than one is that the device may be dual-stack, and we needed a way that the API consumer could specify if they wanted to implement the enhanced QoS for the IPv4 flow, IPv6 flow or both (providing just MSISDN or External Id would imply both if dual-stack).

Obviously this motivation does not apply here, but also I feel we should encourage the API consumer to provide the device identifiers they know, because some are easier to work with than others. It is much easier to determine the roaming status of a MSISDN than a public IP address, and if we restricted the number of device identifiers to one, they might just give us the IP address.

My feeling it is up to the API implementor how they process multiple device identifiers. They can do a consistency check if they want, or just take the identifier most convenient for their implementation. I don't really see that we need to standardise here. Device identifiers are inherently CSP specific.